File: lib/shared/models/product.dart
class Product {
  final String id;
  final String name;
  final String shopName;
  final double price;
  final double? oldPrice;
  final double rating;
  final int reviewsCount;
  final int soldCount;
  final String imagePlaceholder; // Emoji
  final List<String> tags;
  final String description;
  final String category;

  const Product({
    required this.id,
    required this.name,
    required this.shopName,
    required this.price,
    this.oldPrice,
    required this.rating,
    this.reviewsCount = 0,
    this.soldCount = 0,
    required this.imagePlaceholder,
    this.tags = const [],
    this.description = '',
    this.category = 'All',
  });

  factory Product.fromMap(Map<String, dynamic> map) {
    return Product(
      id: map['_id'] ?? map['id'],
      name: map['name'] ?? '',
      shopName: map['seller']?['name'] ?? map['shopName'] ?? '',
      price: (map['price'] ?? 0).toDouble(),
      oldPrice: map['oldPrice']?.toDouble(),
      rating: (map['rating'] ?? 0).toDouble(),
      reviewsCount: map['reviewsCount'] ?? 0,
      soldCount: map['soldCount'] ?? 0,
      imagePlaceholder: (map['images'] != null && map['images'].isNotEmpty) 
        ? map['images'][0] 
        : (map['imagePlaceholder'] ?? '📦'),
      tags: List<String>.from(map['tags'] ?? []),
      description: map['description'] ?? '',
      category: map['category'] ?? 'All',
    );
  }
}

File: backend/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
  name: { type: String, required: true },
  description: { type: String, default: '' },
  price: { type: Number, required: true },
  oldPrice: { type: Number },
  category: { type: String, required: true },
  images: [{ type: String }],
  seller: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  tags: [{ type: String }],
  stock: { type: Number, default: 0 },
  rating: { type: Number, default: 0 },
  reviewsCount: { type: Number, default: 0 },
  soldCount: { type: Number, default: 0 },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Product', productSchema);
